rootio/wbuffer.go

```diff
@@ -94,20 +94,22 @@ func (w *WBuffer) WriteObjectAny(obj Object) error {
 		return w.err
 	}
 
-	//	pos := w.Pos()
-	//	w.WriteU32(0) // placeholder for bytecount.
-	//
-	//	mapsize := len(w.refs)
-	//	w.WriteClass(obj)
+	//pos := w.Pos()
+	w.WriteU32(0) // placeholder for bytecount.
 
+	mapsize := len(w.refs)
+	w.WriteClass(obj)
+	w.w.c += mapsize
```

Comment:
not sure that's really what's needed. (the failing test seems to agree with me :P)